### PR TITLE
[added] number methods less, more, notEqual

### PR DESCRIPTION
**Pull request by A11oW**
Add few new methods for number. This useful use for validate ranges:

```
var yup = require('yup')

var schema = yup.object({
  start: yup.number(),
  end: yup.number().more(yup.ref('start'), 'End must be greater than the start')
})

schema.validate({
  start: 3,
  end: 5,
})
```

Tnx @maullerz for help

### CODE REVIEW

File: src/number.js [hunk 1]
```diff
@@ -9,86 +9,123 @@ let isNaN = value => value != +value
 let isInteger = val => isAbsent(val) || val === (val | 0)
 
 export default function NumberSchema() {
-  if ( !(this instanceof NumberSchema))
-    return new NumberSchema()
+	if ( !(this instanceof NumberSchema))
```
Comment (jquense) [inline src/number.js]: can you fix the tab spacing here? It should be two spaces, no tabs.

**Comment by A11oW** [comment]
I fixed the tab spacing.

File: src/number.js (hunk 1 above)
Comment (A11oW) [inline src/number.js]: I fixed the tab spacing

**Comment by jquense** [comment]
thanks! sorry for the long delay! :|

**Comment by jquense** [comment]
fyi i'm tweaking the names a bit, and I think actually removing `notEqual` since `notOneOf` should already cover that case

**Pull request merged**